Files/Filesystem/CloudDrivesManager.cs

```diff
@@ -0,0 +1,177 @@
+        private List<DriveItem> drivesList = new List<DriveItem>();
+
+        public IReadOnlyList<DriveItem> Drives
+        {
+            get
+            {
+                lock (drivesList)
+                {
+                    return drivesList.ToList().AsReadOnly();
+                }
+            }
+        }
+
+        private async Task EnumerateDrivesAsync()
+        {
+            var cloudProviderController = new CloudProviderController();
+            await cloudProviderController.DetectInstalledCloudProvidersAsync();
+
+            foreach (var provider in cloudProviderController.CloudProviders)
+            {
+                var cloudProviderItem = new DriveItem()
+                {
+                    Text = provider.Name,
+                    Path = provider.SyncFolder,
+                    Type = DriveType.CloudDrive,
+                };
+                lock (drivesList)
+                {
+                    drivesList.Add(cloudProviderItem);
+                }
+            }
+
+            await RefreshUI();
+        }
+
+        public static async Task<CloudDrivesManager> CreateInstance()
+        {
+            var drives = new CloudDrivesManager();
+            await drives.EnumerateDrivesAsync();
+            return drives;
+        }
+
+        private async Task RefreshUI()
+        {
+            try
+            {
+                await SyncSideBarItemsUI();
+            }
+            catch (Exception)       // UI Thread not ready yet, so we defer the pervious operation until it is.
+            {
+                Debug.WriteLine($"RefreshUI Exception");
+                // Defer because UI-thread is not ready yet (and DriveItem requires it?)
+                CoreApplication.MainView.Activated += RefreshUI;
+            }
+        }
+
+        private async void RefreshUI(CoreApplicationView sender, Windows.ApplicationModel.Activation.IActivatedEventArgs args)
+        {
+            await SyncSideBarItemsUI();
+            CoreApplication.MainView.Activated -= RefreshUI;
+        }
+
+        private async Task SyncSideBarItemsUI()
+        {
+            Debug.WriteLine("SyncSideBarItemsUI");
+            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+            {
+                lock (MainPage.SideBarItems)
+                {
+                    var drivesSection = MainPage.SideBarItems.FirstOrDefault(x => x is HeaderTextItem && x.Text == "Cloud Drives");
+
+                    if (drivesSection != null && Drives.Count == 0)
+                    {
+                        //No drives - remove the header
+                        MainPage.SideBarItems.Remove(drivesSection);
+                    }
+
+                    if (drivesSection == null)
+                    {
+                        drivesSection = new HeaderTextItem()
+                        {
+                            Text = "Cloud Drives"
```

Comment:
Is it posible to add this string to the en-US resource file?

Reply:
Of course. Does it matter if a string does not have translations in all of the other resource files?

Reply:
@AndyMDoyle I will take care of generating translations in the other files.

Reply:
Done